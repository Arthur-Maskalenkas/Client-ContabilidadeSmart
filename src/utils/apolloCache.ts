import { InMemoryCache } from '@apollo/client'
import { concatPagination } from '@apollo/client/utilities'

// A query QueryGames retorna -> games <-, e é isso que o cache vai acumular games

// ** fluxo:
// > Eu pego uma lista de free (retorna games assim como varias outras query)
// > o cache fica [game1,game2,game3]
// > Assim que outra parte usa alguma query que retorne game, ela vai receber independente de tudo [game1,game2,game3], e não [game4,game5,game6]
// **

// Para quebrar o fluxo, eu separei o cacheamento por 'where' e 'sort'. Assim fica algo como: (caso puxe 3 games de 3 queries diferentes)

//**
// > games: {"where: {"release_date_lte":"2021-01-29"}, "sort":"release_date:desc"}: [] 8 items"}
// > games: {"where: {"release_date_gt":"2021-01-29"}, "sort":"release_date:asc"}: [] 8 items"}
// > games: {"where: {"price": 0}}: [] 8 items
// > Antes ficaria tudo em games com 24 items
//**
export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        posts: concatPagination(['where', 'sort'])
      }
    }
  }
})
