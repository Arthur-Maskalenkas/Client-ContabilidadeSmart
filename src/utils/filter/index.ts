import { ParsedUrlQueryInput } from 'querystring'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
}

export const parseQueryStringToWhere = ({ queryString }: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  // queryString = {platforms: ['windows','linux']}
  // filterItems = {name: 'platforms', type: 'checkbox'}
  Object.keys(queryString).forEach((key) => {
    // Pegando o valor e colocando dentro do objeto
    obj[key] = { slug_contains: queryString[key] }
  })

  return obj
}
