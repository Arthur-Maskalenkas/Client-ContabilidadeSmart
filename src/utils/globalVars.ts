export const IMAGE_NOT_FOUND =
  'https://res.cloudinary.com/dezwlfeyb/image/upload/v1635853776/102-022_800x_ebxk1t.jpg'

export const LINK_SITE = 'localhost:3000'

// --  API

export const API_LINKEDIN =
  'https://www.linkedin.com/shareArticle?mini=true&url='

export const API_FACEBOOK = 'https://www.facebook.com/sharer/sharer.php?u='

export const API_WHATSAPP = 'https://api.whatsapp.com/send?text='

// --  API
