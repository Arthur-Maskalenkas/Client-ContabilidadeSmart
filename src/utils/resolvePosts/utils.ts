export const operations = {
  nextPage: nextPageOperation(),
  backPage: backPageOperation()
}

type variablesType = {
  limit?: number
  start?: number
}

export type FetchMoreProps = {
  variables: variablesType
}
