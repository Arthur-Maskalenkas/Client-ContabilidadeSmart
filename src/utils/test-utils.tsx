import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const customRender = (ui: ReactElement, { ...renderOptions }: CustomRenderProps = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

// Tudo que eu chamar, vai vim dessa pasta aqui
export * from '@testing-library/react'
export { customRender as render }
