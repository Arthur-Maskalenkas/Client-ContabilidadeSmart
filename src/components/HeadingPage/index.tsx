import { Container } from 'components/Container'
import * as S from './styles'

export type HeadingPageProps = {
  title: string
}

const HeadingPage = ({ title }: HeadingPageProps) => (
  <S.Wrapper>
    <Container>
      <S.ContentWrapper>
        <S.Title>{title}</S.Title>

        <S.linhas>
          <S.linha />
          <S.linha />
          <S.linha />
        </S.linhas>
      </S.ContentWrapper>
    </Container>
  </S.Wrapper>
)

export default HeadingPage
