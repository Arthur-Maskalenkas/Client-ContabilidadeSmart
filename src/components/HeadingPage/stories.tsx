import { Story, Meta } from '@storybook/react'
import HeadingPage, { HeadingPageProps } from '.'

import { MINIMAL_VIEWPORTS } from '@storybook/addon-viewport'

export default {
  title: 'HeadingPage',
  component: HeadingPage
} as Meta

export const Default: Story<HeadingPageProps> = (args) => <HeadingPage {...args} />

Default.args = { title: 'Saques de FGTS e recebimento de benefício' }

Default.parameters = {
  backgrounds: {
    default: 'mm-light'
  }
}

export const Mobile: Story<HeadingPageProps> = (args) => <HeadingPage {...args} />

Mobile.args = { title: 'Saques de FGTS e recebimento de benefício' }

Mobile.parameters = {
  backgrounds: {
    default: 'mm-light'
  },
  viewport: {
    viewports: MINIMAL_VIEWPORTS,
    defaultViewport: 'mobile1'
  }
}
