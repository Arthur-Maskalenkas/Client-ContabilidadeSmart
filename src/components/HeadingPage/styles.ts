import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { Container } from 'components/Container'

import * as LogoStyles from 'components/Logo/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    background-color: ${theme.colors.darkGray};
  `}
`

export const ContentWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    height: 55rem;

    ${LogoStyles.Wrapper}, ${Title} {
      position: relative;
      z-index: ${theme.layers.base};
    }

    ${media.greaterThan('large')`
      height: 35rem;
    `}
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xlarge};
    font-weight: ${theme.font.family.poppins.weight.bold};
    color: ${theme.colors.white};

    ${media.greaterThan('large')`
      margin-top: 20rem;
      margin-left: 3.2rem;
      width: 50%;
    `};
  `}
`

// Animação
export const linhas = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 100%;
  margin: auto;
  z-index: 1;
`
export const linha = styled.div`
  ${({ theme }) => css`
    position: absolute;
    width: 0.1rem;
    height: 100%;
    top: 0;
    left: 50%;
    background: ${theme.colors.bordasCinzas};
    overflow: hidden;

    &:after {
      content: '';
      display: block;
      position: absolute;
      height: 15rem;
      width: 100%;
      top: -50%;
      left: 0;
      background: ${theme.colors.white};

      animation: run 4s 0s infinite;
      animation-fill-mode: forwards;
      animation-timing-function: cubic-bezier(0.4, 0.26, 0, 0.97);
    }

    &:nth-child(1) {
      margin-left: -25%;
    }

    &::nth-child(1):after {
      animation-delay: 2s;
    }

    &:nth-child(3) {
      margin-left: 25%;
    }

    &:nth-child(3):after {
      animation-delay: 2.5s;
    }

    @keyframes run {
      0% {
        top: -50%;
      }
      100% {
        top: 110%;
      }
    }
  `}
`
