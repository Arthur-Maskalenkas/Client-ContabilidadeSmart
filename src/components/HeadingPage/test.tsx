import { render, screen } from 'utils/test-utils'

import HeadingPage, { HeadingPageProps } from '.'

const props: HeadingPageProps = { title: 'Um titulo' }

describe('<HeadingPage />', () => {
  it('vai renderizar o componente', () => {
    const { container } = render(<HeadingPage {...props} />)

    expect(screen.getByRole('heading', { name: /Um titulo/i })).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
