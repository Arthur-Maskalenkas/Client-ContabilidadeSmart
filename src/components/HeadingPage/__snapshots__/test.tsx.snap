// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadingPage /> vai renderizar o componente 1`] = `
.c1 {
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(1.5rem);
  padding-right: calc(1.5rem);
}

.c0 {
  background-color: #3F4350;
  overflow: hidden;
}

.c2 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 55rem;
}

.c3 {
  font-size: 4.2rem;
  font-weight: 600;
  color: #FFFFFF;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 100%;
  margin: auto;
  z-index: 1;
}

.c5 {
  position: absolute;
  width: 0.1rem;
  height: 100%;
  top: 0;
  left: 50%;
  background: #535662;
  overflow: hidden;
}

.c5:after {
  content: '';
  display: block;
  position: absolute;
  height: 15rem;
  width: 100%;
  top: -50%;
  left: 0;
  background: #FFFFFF;
  -webkit-animation: run 4s 0s infinite;
  animation: run 4s 0s infinite;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-timing-function: cubic-bezier(0.4,0.26,0,0.97);
  animation-timing-function: cubic-bezier(0.4,0.26,0,0.97);
}

.c5:nth-child(1) {
  margin-left: -25%;
}

.c5::nth-child(1):after {
  -webkit-animation-delay: 2s;
  animation-delay: 2s;
}

.c5:nth-child(3) {
  margin-left: 25%;
}

.c5:nth-child(3):after {
  -webkit-animation-delay: 2.5s;
  animation-delay: 2.5s;
}

@media (min-width:1170px) {
  .c2 {
    height: 35rem;
  }
}

@media (min-width:1170px) {
  .c3 {
    margin-top: 20rem;
    margin-left: 3.2rem;
    width: 50%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h1
          class="c3"
        >
          Um titulo
        </h1>
        <div
          class="c4"
        >
          <div
            class="c5"
          />
          <div
            class="c5"
          />
          <div
            class="c5"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
